src/main/java/bowling/domain/frame/Frame.java

```diff
@@ -12,64 +12,113 @@
 import static bowling.Messages.WARNING_FRAME_NOT_ALLOWED_SUM;
 
 public class Frame {
+    private static final RandomGenerator RANDOM_GENERATOR = new RandomGenerator();
     private static final int SCORE_TEN = 10;
     private static final int SCORE_ZERO = 0;
     private static final int LAST_FRAME_ID = 10;
+    private static final int FIRST_FRAME_ID = 1;
+    private static final int FRAME_ID_ZERO = 0;
+    private static final int FRAME_ID_NINETH = 9;
+    private static final int INCREMENT_FOR_NEXT_ID = 1;
 
-    private int frameId = 1;
+    private int frameId;
     private Points points;
 
-    public Frame(int firstPoint, int secondPoint) {
-        validateSecondWhenFirstTen(firstPoint, secondPoint);
-        validateSumIsLessThanTen(firstPoint, secondPoint);
-        points = Points.of(firstPoint, secondPoint);
+    public Frame(int prevFrameId, int firstPoint, int secondPoint) {
+        setUpForConstructor(prevFrameId, firstPoint, secondPoint);
+
+        addPointForSpare();
+        addPointsForStrike();
```

Comment:
```suggestion
        validate(first, second);
        
        this.frameId = prevFrameId + INCREMENT_FOR_NEXT_ID;
        points = Points.of(first, second, needOnePoint(), needTwoPoint());
```

외부에서 Points의 값을 set하다보니 unmodifiable 라이브러리 사용 시점이 분기되고 점차 데이터 흐름에 따라 절차적으로 설계하게 되는데요. Points에 대한 값을 생성하는 역할은 Points 객체에 위임하는 것은 어떨까요